### PR TITLE
update CI and Changelog

### PR DESCRIPTION
**Pull request by sateeshperi**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,6 +35,7 @@ jobs:
           - "24.04.2"
           - "latest-everything"
         shard: [1, 2, 3, 4, 5]
+        filter: [process, workflow]
```
Comment (GallVp) [inline .github/workflows/ci.yml:38]: The pipeline-level tests are not included?

Comment (sateeshperi) [reply .github/workflows/ci.yml:38]: ah good catch! should be pipeline and workflow

Comment (GallVp) [reply .github/workflows/ci.yml]: And process? I think the idea is to test everything?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -49,6 +50,12 @@ jobs:
             profile: "conda"
           - isMaster: false
             profile: "singularity"
+          - profile: "conda"
```
Comment (GallVp) [inline .github/workflows/ci.yml:53]: If these workflows can't run with conda, I assume the whole pipeline can also be not run with conda when using bwameth. Is this documented somewhere?

Comment (sateeshperi) [reply .github/workflows/ci.yml:53]: actually the pipeline can be run using conda. It seems there is only an issue when running conda on CI for pipeline tests. see [this issue](https://github.com/nf-core/modules/issues/6664)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sateeshperi) [reply .github/workflows/ci.yml]: I think its gonna overload the CI runners with too many tests to run

Comment (sateeshperi) [reply .github/workflows/ci.yml]: I think we have to bank on the fact that the modules were/are tested in the modules repo and the pipeline CI runs only pipeline and workflow tests

Comment (sateeshperi) [reply .github/workflows/ci.yml]: you should bring this up in the `#nf-test` channel and we can discuss/vote there

Comment (GallVp) [reply .github/workflows/ci.yml]: What if there are local modules with tests? Moreover, the tests are only triggered for the changed modules.

If there are no local modules or functions with tests, let's skip `process`.

One more thing. If the nextflow.config changes or something from the `assets` directory changes, will the pipeline-level tests be triggered? I like the CI of nf-core/rnaseq which checks for these changes.

**Review by GallVp — APPROVED** [review]

**Comment by GallVp** [comment]
LGTM!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sateeshperi) [reply .github/workflows/ci.yml]: plz bring this up in the `nf-test` channel and lets discuss there and get some consensus

**Pull request merged**